pkg/logs/diagnostic/message_receiver.go

```diff
@@ -0,0 +1,133 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package diagnostic
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/logs/config"
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+)
+
+// MessageReceiver interface to handle messages for diagnostics
+type MessageReceiver interface {
+	HandleMessage(message.Message)
+}
+
+// BufferedMessageReceiver handles in coming log messages and makes them available for diagnostics
+type BufferedMessageReceiver struct {
+	inputChan chan message.Message
+	enabled   bool
+	m         sync.RWMutex
+}
+
+// Filters for processing log messages
+type Filters struct {
+	Name   string `json:"name"`
+	Type   string `json:"type"`
+	Source string `json:"source"`
+}
+
+// NewBufferedMessageReceiver creates a new MessageReceiver
+func NewBufferedMessageReceiver() *BufferedMessageReceiver {
+	return &BufferedMessageReceiver{
+		inputChan: make(chan message.Message, config.ChanSize),
+	}
+}
+
+// Start opens new input channel
+func (b *BufferedMessageReceiver) Start() {
+	b.inputChan = make(chan message.Message, config.ChanSize)
```

Comment:
Yes - however the logs agent is restartable. Since `BufferedMessageReceiver` is kept around after initialization, I made it conform to `Startable` and `Stoppable` so it can startup and teardown with the rest of the agent. 
See: https://github.com/DataDog/datadog-agent/pull/7163/files#diff-b553315cdfb35d2a614d3feb241bf3a29d999966b125c3d54c59fca8ad415f27R117